File: js/5052.js
const fs = require('fs');

const inputs = fs.readFileSync('example.txt').toString().trim().split('\n');

let idx = 0;
const getLine = () => inputs[idx++];

let T = +getLine();
const ans = [];

while (T--) {
  const N = +getLine();
  const arr = Array(N)
    .fill()
    .map(() => getLine())
    .sort();

  let preL = 0;
  let a = 'YES';

  for (let i = 0; i < arr.length; i++) {
    if (arr[i].slice(0, preL) === arr[i - 1]) {
      a = 'NO';
      break;
    }
    preL = arr[i].length;
  }
  ans.push(a);
}

console.log(ans.join('\n'));

File: c++/5052.cpp
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int main() {
  freopen("example.txt", "r", stdin);

  int T;
  cin >> T;

  while (T--) {
    int N;
    cin >> N;
    vector<string> v;

    while (N--) {
      string s;
      cin >> s;
      v.push_back(s);
    }

    sort(v.begin(), v.end());

    string preS = v[0];
    int preL = preS.length();
    string ans = "YES";

    for (int i = 1; i < v.size(); i++) {
      if (v[i].substr(0, preL) == preS) {
        ans = "NO";
        break;
      }
      preS = v[i];
      preL = v[i].length();
    }
    cout << ans << '\n';
  }

  return 0;
}